src/v/storage/spill_key_index.h

```diff
@@ -87,11 +92,12 @@ class spill_key_index final : public compacted_index_writer::impl {
     ss::future<> spill(compacted_index::entry_type, bytes_view, value_type);
 
     storage::debug_sanitize_files _debug;
+    storage_resources& _resources;
     ss::io_priority_class _pc;
     bool _truncate;
     std::optional<segment_appender> _appender;
     underlying_t _midx;
-    size_t _max_mem;
+    size_t _max_mem{512_KiB};
```

Comment:
is this because there is a delay for storage resources to propagate a "good" value?

Reply:
No, this is the same fixed value that was used before this PR, but written into the class definition instead of hardcoded into the util method that constructed this object.

Currently spill_key_index takes the resources object so that it can pass it through to segment_appender, but doesn't actually call into it to pick a max_mem parameter -- seen comment below about followup.